daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/Preprocessor.scala

```diff
@@ -184,7 +184,11 @@ private[engine] final class Preprocessor(
       tx: SubmittedTransaction
   ): Result[ImmArray[speedy.Command]] =
     safelyRun(
-      getDependencies(List.empty, tx.rootNodes.toList.map(_.templateId))
+      getDependencies(
+        List.empty,
+        tx.rootNodes.toList.map(_.templateId)
+          ++ tx.byInterfaceNodes.toList.map(_.templateId),
+      )
```

Comment:
Since `tx.byInterfaceNodes` is traversing all the nodes, it is problably simpler to just get all the template Id of all the nodes.
```suggestion
  getDependencies(
        List.empty,
        tx.nodes.values.collect { case node: Node.Action => node.templateId }.toList)
  )
```

Reply:
Won't this slow things down for transactions that don't use interfaces?

Reply:
I do like the simplicity of just fetching all templateIds, though.

Reply:
> Won't this slow things down for transactions that don't use interfaces?

After looking at `getDependencies`, you may be right. 
However I do not understand why I made it so complicate.